bubble_sort.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+def bubble_sort(list)
+  iter = list.size - 2
+  max_iter = list.size - 2
+  while max_iter >= 2
+    swap = false
+    0.upto(iter) do |i|
+      if list[i] > list[i + 1]
+        list[i], list[i + 1] = list[i + 1], list[i]
+        swap = true
+      end
+    end
+    max_iter -= 1
+    puts 'max_iter: ', max_iter
+    break unless swap
+  end
+  list
+end
```

Comment:
**[REQUIRED]**
After testing the method using `[10, 9, 8, 7, 6, 5, 4, 3, 2, 1]` , I got `[3, 2, 1, 4, 5, 6, 7, 8, 9, 10]`. This is the expected output `[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]`. Let's fix this.